control/main.py

```diff
@@ -0,0 +1,194 @@
+"""Main program for autonomous drone flight and sensor data logging.
+
+Uses the ttyO4 connection (UART4) on the beaglebone to connect to a Pixhawk flight controller.
+Uses the ttyO1 connection (UART1) to connect to a xBee wireless communication module.
+Uses tcp://localhost:5555 to connect to zmq server for reading i2c sensor data.
+Installing the appropriate main.service file allows for this to start on boot.
+"""
+
+import logging
+import sys
+import time
+import dronekit
+from control.communication import Communication
+from control.controller import Controller
+from control.i2cdataclient import I2cDataClient
+from control.gps import get_location_offset, get_distance, get_relative_from_location
+from control.helper import location_global_relative_to_gps_reading, gps_reading_to_location_global
+
+MAX_RADIUS = 50
```

Comment:
Comment above these saying something like `# Set max and min values just to be safe` so anyone looking at this later knows there isn't a really legit reason for these values.